### PR TITLE
REG-1134: refactor PAYE to share common 'admin data' approach with VAT

### PR DESCRIPTION
**Pull request by nigelhp**
Retrieval of PAYE now shares the same approach as previously implemented for VAT.
Acceptance tests now cover the core behaviour.

Further refactoring to provide a common implementation for 'admin data'.  Subsequent stories to refactor the Companies House endpoint should now require little more than adding acceptance tests, and instantiating the necessary components in the module.

### CODE REVIEW

File: test/controllers/v1/PayeRoutingSpec.scala [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  private trait Fixture {
+    val ValidPayeRef = "125H7A71620"
+    val ValidPeriod = "201804"
+  }
+
+  "A request to retrieve a PAYE unit by PAYE reference and period" - {
+    "is rejected when" - {
+      "the VAT reference" - {
+        "has fewer than four characters" in new Fixture {
```
Comment (ameenhaq) [inline test/controllers/v1/PayeRoutingSpec.scala]: Can we double check PAYE minimum length is 4 - I recall there was confusion with the minimum being either 4 or 6

File: test/support/sample/SampleLinkedUnit.scala [hunk 2]
```diff
@@ -13,13 +13,23 @@ object SampleLinkedUnit {
...
+  def forPaye(period: Period, payeRef: PayeRef): LinkedUnit =
+    LinkedUnit(
+      UnitId(payeRef.value),
+      UnitType.PayAsYouEarn,
+      period,
+      parents = Some(Map(UnitType.Enterprise -> UnitId("1234567890"))),
+      children = None,
+      vars = SamplePaye.asJson(payeRef)
     )
 }
```
Comment (ameenhaq) [inline test/support/sample/SampleLinkedUnit.scala]: Could we not create a shared method between these and then implement it with forVat and forPaye

File: test/controllers/v1/PayeRoutingSpec.scala (hunk 1 above)
Comment (nigelhp) [reply test/controllers/v1/PayeRoutingSpec.scala]: This has been confirmed by the product owner in relation to historic data that we have reveived.

File: test/support/sample/SampleLinkedUnit.scala (hunk 2 above)
Comment (nigelhp) [reply test/support/sample/SampleLinkedUnit.scala]: Following the refactoring to more generic services, these are no longer used.  I will remove them on a subsequent branch.

**Pull request merged**